src/benchmark.cc

```diff
@@ -813,6 +817,29 @@ int InitializeStreams() {
 
 }  // end namespace internal
 
+bool MaybeReenterWithoutASLR(int /*argc*/, char** argv) {
+  // On e.g. Hexagon simulator, argv may be NULL.
+  if (!argv) return false;
+
+#ifdef BENCHMARK_OS_LINUX
+  const auto curr_personality = personality(0xffffffff);
+  BM_CHECK(curr_personality != -1)
```

Comment:
this will only fail in DEBUG. should it be a stronger check?

Reply:
`man 2 personality` says:
```
       If persona is not 0xffffffff, then personality() sets the caller's execution domain to the value specified by persona.  Specifying persona as 0xffffffff provides a way of retrieving the current persona without changing it.
```
```
RETURN VALUE
       On success, the previous persona is returned.  On error, -1 is returned, and errno is set to indicate the error.

ERRORS
       EINVAL The kernel was unable to change the personality.
```
So generally it shouldn't just fail. But sure, let's do that.